buildchain/constants.py

```diff
@@ -0,0 +1,88 @@
+# coding: utf-8
+
+
+"""Useful global constants, used hither and yon."""
+
+
+import os
+from pathlib import Path
+from typing import Tuple
+
+from buildchain import ROOT  # Re-export ROOT through this module.
+from buildchain import config
+
+
+# Max length of a "command".
+# (used in task display, for a nice aligned output).
+CMD_WIDTH : int = 12
+
+# URLs of the main container repositories.
+GOOGLE_REGISTRY : str = 'k8s.gcr.io'
+DOCKER_REGISTRY : str = 'docker.io/library'
+
+# Paths {{{
+
+# Root of the generated ISO.
+ISO_ROOT : Path = config.BUILD_ROOT/'root'
+# Root of the repositories on the ISO.
+REPO_ROOT : Path = ISO_ROOT/'packages'
+# Root for the packages that we build ourself.
+PKG_ROOT : Path = config.BUILD_ROOT/'packages'
+
+# }}}
+# Container options {{{
+
+BIND_RO_MOUNT_FMT : str = 'type=bind,source={src},destination={dst},ro'
+BUILDER_RPMLINTRC_MOUNT : str = BIND_RO_MOUNT_FMT.format(
+    src=ROOT/'packages'/'rpmlintrc',
+    dst='/rpmbuild/rpmlintrc'
+)
+BUILDER_ENTRYPOINT_MOUNT : str = BIND_RO_MOUNT_FMT.format(
+    src=ROOT/'packages'/'entrypoint.sh',
+    dst='/entrypoint.sh'
+)
+BUILDER_BASIC_CMD : Tuple[str, ...] = (
+    config.DOCKER, 'run',
+    '--env', 'TARGET_UID={}'.format(os.geteuid()),
+    '--env', 'TARGET_GID={}'.format(os.getegid()),
+    '--hostname', 'build',
+    '--mount', 'type=tmpfs,destination=/tmp',
+    '--mount', BUILDER_ENTRYPOINT_MOUNT,
+    '--rm',
+)
+
+# }}}
+# Versions {{{
+
+K8S_VERSION  : str = '1.11.7'
+SALT_VERSION : str = '2018.3.3'
+
+
+def load_version_information() -> None:
+    """Load version information from `VERSION`."""
+    to_update = {
+        'VERSION_MAJOR', 'VERSION_MINOR', 'VERSION_PATCH', 'VERSION_SUFFIX'
+    }
+    with VERSION_FILE.open('r', encoding='utf-8') as fp:
+        for line in fp:
+            name, _, value = line.strip().partition('=')
+            # Don't overwrite random variables by trusting an external file.
+            if name in to_update:
+                globals()[name.strip()] = value.strip()
```

Comment:
Wouldn't it be nicer if this returns a tuple, which gets then bound in the local scope?

```python
def load_version_information():
    ...
    return (major, minor, patch, suffix)

VERSION_MAJOR, VERSION_MINOR, VERSION_PATCH, VERSION_SUFFIX = load_version_information()
```

Reply:
Not sure it's nice, the `...` will probably be more complicated than the actual `…`.

So far I'm getting

```
def load_version_information() -> Tuple[str, str, str, str]:
    """Load version information from `VERSION`."""
    to_update = {
            'VERSION_MAJOR': '',
            'VERSION_MINOR': '',
            'VERSION_PATCH': '',
            'VERSION_SUFFIX': '',
    }
    with VERSION_FILE.open('r', encoding='utf-8') as fp:
        for line in fp:
            name, _, value = line.strip().partition('=')
            # Don't overwrite random variables by trusting an external file.
            if name.strip() in to_update:
                to_update[name.strip()] = value.strip()
        return (
            to_update['VERSION_MAJOR'],
            to_update['VERSION_MINOR'],
            to_update['VERSION_PATCH'],
            to_update['VERSION_SUFFIX']
        )

VERSION_FILE : Path= ROOT/'VERSION'

# Metalk8s version.
VERSION_MAJOR, VERSION_MINOR, VERSION_PATCH, VERSION_SUFFIX = load_version_information()
```

Not convinced by the gain (but maybe there is a simpler body for `load_version_information`).